sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/InMemoryPartitionManager.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   A non-volatile in-memory storage service that keeps track of checkpoints and ownership.  It must
+    ///   be used with a single <see cref="EventProcessor"/> instance.
+    /// </summary>
+    ///
+    public class InMemoryPartitionManager : PartitionManager
+    {
+        /// <summary>The primitive for synchronizing access during ownership claim.</summary>
+        private readonly object OwnershipClaimLock = new object();
+
+        /// <summary>The set of stored checkpoints.  Partition ids are used as keys.</summary>
+        private ConcurrentDictionary<string, Checkpoint> Checkpoints;
+
+        /// <summary>The set of stored ownership.  Partition ids are used as keys.</summary>
+        private ConcurrentDictionary<string, PartitionOwnership> Ownership;
```

Comment:
Given that you lock when modifying this, I don't know that you want to use a ConcurrentDictionary here, rather than an actual dictionary.

Reply:
Agreed.